test/modal.test.js

```diff
@@ -79,45 +83,46 @@ describe('Modal', () => {
         }}
       />
     )
-    expect(
-      wrapper.find('.modal-fade-screen.custom.modal-is-open').exists()
-    ).toEqual(true)
+    const overlay = screen.getByRole('dialog').parentElement
+    expect(overlay).toHaveClass('modal-fade-screen', 'custom', 'modal-is-open')
   })
 
   describe('when preventClose=true', () => {
     test('hides close button', () => {
-      const wrapper = mount(<Modal preventClose={true} onClose={noop} />)
-      expect(wrapper.find('.modal-close').exists()).toEqual(false)
+      render(<Modal preventClose={true} onClose={noop} />)
+      expect(screen.queryByRole('button', { name: 'Close Modal' })).not.toBeInTheDocument()
     })
 
-    test('does not close by escape key', () => {
+    test('does not close by escape key', async () => {
       const onClose = jest.fn()
-      const wrapper = mount(<Modal preventClose={true} onClose={onClose} />)
-      wrapper.find('.modal-content').simulate('keydown', { keyCode: 27 })
+      const user = userEvent.setup()
+      render(<Modal preventClose={true} onClose={onClose} />)
+
+      await user.keyboard('{Escape}')
       expect(onClose).not.toHaveBeenCalled()
     })
```

Comment:
Should we add tests for the default case, that it does close on Esc and on overlay click? I realize that it's not part of the original tests but it would make them more reliable.

Reply:
Yeah, that makes sense to me. Will add now